sw/vendor/riscv_compliance/riscv-target/opentitan/device/rv32imc/handler.S

```diff
@@ -0,0 +1,38 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+.section .text.trap;
+.align  4;
+
+_trap_start:
+    j _trap_exception
+    j _int_exc
+    j _int_exc
+    j _int_exc
+    j _int_exc
+    j _int_exc
+    j _int_exc
+    j _int_exc
+    j _int_exc
+    j _int_exc
+    j _int_exc
+    j _int_exc
+    j _int_exc
+
+
+// This could be exception or user interrupt
+// 0xb is the environment call to indicate the end
+_trap_exception:
+    csrr a0, mcause
+    addi a1, zero, 0xb
+    beq a0, a1, 1f
+    la a1, begin_signature
+    sw a0, 0(a1)
```

Comment:
Write `sw a0, begin_signature, a1`. This will write to the value pointed to by the symbol `begin_signature`, using `a1` as scratch space for the address.